### PR TITLE
Feat/layout engine

### PR DESCRIPTION
**Pull request by primequantuM4**
### Added
- `LayoutEngine` to compute positions for child components based on direction and available space.
- `PositionedComponent` to associate layout bounds with renderable components.
- `Row` and `Column` components for horizontal and vertical composition.
- Unit tests for `LayoutEngine` covering relative positioning and edge cases.

### Notes
- `fitWidth` and `fitHeight` left unimplemented pending dynamic sizing logic.
- Absolute-positioned children are currently ignored by the layout engine.

### CODE REVIEW

File: lib/components/row.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  Row({required this.children});
+
+  @override
+  Size measure(Size maxSize) {
+    int maxHeight = 0;
+    int totalWidth = 0;
+
+    for (final child in children) {
+      if (child.position?.positionType == PositionType.absolute) continue;
```
Comment (TheComputerM) [inline lib/components/row.dart:23]: Why is child.position optional?

Comment (primequantuM4) [reply lib/components/row.dart:23]: Because the user might give their own custom position the attribute position in a component is right now nullable since when users call a component class like text component it shouldnt force them to specify position. But it also should allow them to override the custom position.
